### PR TITLE
Ahead/behind in branch details

### PR DESCRIPTION
**Pull request by pfeuffer**
## Proposed changes

Adds the ahead/behind count to the branch details view. 

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [ ] Related issues linked to PR if existing and labels set
- [x] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Feature has been tested with different permissions
- [x] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [x] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [x] All new code/logic is implemented on the right spot / "Should this be done here?"
- [x] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [x] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [x] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [x] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: docs/en/user/repo/branches.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ New branches can be created with the "Create Branch" form. There, you have to ch
 ![Create Branch](assets/repository-create-branch.png)
 
 ### Branch Details Page
-This page shows some commands to work with the branch on the command line.
+This page shows some commands to work with the branch on the command line. The ahead/behind commits can be found directly
+beneath the name of the branch.
```
Comment (matthiasthieroff) [inline docs/en/user/repo/branches.md:26]: This is a little bit ambiguous. I would prefer something like "The number of commits ahead/behind can..."

File: scm-ui/ui-api/src/branches.ts [hunk 2]
```diff
@@ -111,6 +113,12 @@ export const useBranchDetailsCollection = (repository: Repository, branches: Bra
           return undefined;
         }
         return allPages.length;
+      },
+      onSuccess: data => {
+        const flatMap = data.pages.flatMap(d => d._embedded?.branchDetails);
```
Comment (matthiasthieroff) [inline scm-ui/ui-api/src/branches.ts:118]:
```suggestion
        const details = data.pages.flatMap(d => d._embedded?.branchDetails);
```

**Review by matthiasthieroff — COMMENTED** [review]
Just two small improvements, the rest looks good.

File: scm-ui/ui-api/src/branches.ts (hunk 2 above)
Comment (pfeuffer) [reply scm-ui/ui-api/src/branches.ts]: Right, this was just for debugging purposes. I removed the variable completely.

**Pull request merged**